### DISCUSSION TITLE
How can i write maths in slate. Only this feature can make slate a complete editor? This feautre is essential for my application. Any suggestion

### DISCUSSION THREAD

**Discussion by roker15** [ianstormtaylor/slate#4596] [General]
How can i write maths in slate. Only this feature can make slate a complete editor? This feautre is essential for my application. Any suggestion

**Comment by dylans** [comment]
From https://docs.slatejs.org/general/resources there is a link to https://cocalc.com/ which is a commercial product based on Slate that shows it is possible.

There have been various discussions over time when people ask about math in slatejs, e.g. https://stackoverflow.com/questions/52138467/adding-inline-blocksmath-equation-in-slate-js  , https://github.com/ianstormtaylor/slate/issues/1913 , 

There is also a stub for a slate-math project at https://github.com/slate-math/slate-math which doesn't seem active but it might be worth reaching out to them.

Or perhaps something like https://github.com/mathquill/mathquill would be used as void elements within Slate?

**Reply by roker15** [reply]
Why don't you write official plugin for katex? Is there plan for this in near future?

**Reply by roker15** [reply]
> Or perhaps something like https://github.com/mathquill/mathquill would be used as void elements within Slate?

Can you elaborate this how can we use this in slate? Do i need to add separate button in toolbar? please provide full code, this feature is must for my project

**Reply by TheSpyder** [reply]
This is an open source project run mostly by volunteers. Slate is a framework to build editors - an official plugin for something as specific as writing maths inside the editor isn't really what we do.

The expectation is someone integrating Slate should be able to implement such features themselves. If you need more help than we can provide, there are plenty of commercial products built around Slate (such as cocalc which @dylans mentioned).

**Reply by roker15** [reply]
> This is an open source project run mostly by volunteers. Slate is a framework to build editors - an official plugin for something as specific as writing maths inside the editor isn't really what we do.
> 
> The expectation is someone integrating Slate should be able to implement such features themselves. If you need more help than we can provide, there are plenty of commercial products built around Slate (such as cocalc which @dylans mentioned).

Every editor in market vacillate between table and formulas ... Every other things are available with every editor. So please at least provide concrete implementation of table and formulas.
